app/assets/javascripts/tylium/modules/server_auto_save.js

```diff
@@ -0,0 +1,74 @@
+// Server Auto Save module is initialized on elements with
+// data-behavior=server-auto-save. This module creates an action cable
+// subscription bound to the editor it's initialized from. The socket will
+// accept content for resources to update, and create revisions from. It will
+// receive timestamps back from the server to update the `original_updated_at`
+// field which helps us with conflict resolution.
+(function($, window) {
+  function ServerAutoSave(form) {
+    this.form = form;
+
+    this._doneTypingInterval = 500;
```

Comment:
I was going to suggest to move this interval to a `<meta>` somewhere so we can change the typing timeout value all over the place in one go.

However, now I'm thinking, shouldn't this be something that's triggered from the Editor?

Otherwise we have two separate JS modules (the Editor and ServerAutoSave), checking for a single input, and firing actions and Ajax/ActionCable calls on intervals after the user stops typing.

It's not the end of the world, but it seems there's something weird about this being an "editor functionality" that's separate from the editor?

Maybe having all the editor preview, field-source sync events and a couple of millis later having the auto-save events fire is not such a big deal?

Reply:
This discussion is happening on another platform. Marking this as resolved because we will move forward with the current implementation ensuring to revisit the same methods before final merge.